public/app/features/dashboard-scene/panel-edit/PanelDataPane/NewAlertRuleButton.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { useLocation } from 'react-router-dom';
+import { useAsync } from 'react-use';
+
+import { urlUtil } from '@grafana/data';
+import { locationService, logInfo } from '@grafana/runtime';
+import { SceneQueryRunner, VizPanel } from '@grafana/scenes';
+import { Alert, Button } from '@grafana/ui';
+import { LogMessages } from 'app/features/alerting/unified/Analytics';
+import { scenesPanelToRuleFormValues } from 'app/features/alerting/unified/utils/rule-form';
+
+import { DashboardScene } from '../../scene/DashboardScene';
+
+interface ScenesProps {
+  panel: VizPanel;
+  queryRunner: SceneQueryRunner;
+  dashboard: DashboardScene;
+  className?: string;
+}
```

Comment:
This is the type I was refering to in my previous comment ref simplifying the API.